### PR TITLE
[Monitor] Fix issue #41377

### PR DESCRIPTION
**Pull request by HarveyLink**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

**Comment by HarveyLink** [comment]
https://github.com/Azure/azure-sdk-for-net/issues/41377

File: sdk/monitor/Azure.ResourceManager.Monitor/src/Customized/Models/MonitorConditionOperator.Serialization.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    {
+        // Fix for issue: https://github.com/Azure/azure-sdk-for-net/issues/41377
+        // The swagger has enum value `Equals`, but the service response is `Equal`
+        public static string ToSerialString(this MonitorConditionOperator value) => value switch
+        {
+            MonitorConditionOperator.GreaterThan => "GreaterThan",
+            MonitorConditionOperator.GreaterThanOrEqual => "GreaterThanOrEqual",
+            MonitorConditionOperator.LessThan => "LessThan",
+            MonitorConditionOperator.LessThanOrEqual => "LessThanOrEqual",
+            MonitorConditionOperator.EqualsValue => "Equal",
```
Comment (ArcturusZhang) [inline sdk/monitor/Azure.ResourceManager.Monitor/src/Customized/Models/MonitorConditionOperator.Serialization.cs:20]: please add another note here as well

File: sdk/monitor/Azure.ResourceManager.Monitor/src/Customized/Models/MonitorConditionOperator.Serialization.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+            _ => throw new ArgumentOutOfRangeException(nameof(value), value, "Unknown MonitorConditionOperator value.")
+        };
+
+        public static MonitorConditionOperator ToMonitorConditionOperator(this string value)
+        {
+            if (StringComparer.OrdinalIgnoreCase.Equals(value, "GreaterThan")) return MonitorConditionOperator.GreaterThan;
+            if (StringComparer.OrdinalIgnoreCase.Equals(value, "GreaterThanOrEqual")) return MonitorConditionOperator.GreaterThanOrEqual;
+            if (StringComparer.OrdinalIgnoreCase.Equals(value, "LessThan")) return MonitorConditionOperator.LessThan;
+            if (StringComparer.OrdinalIgnoreCase.Equals(value, "LessThanOrEqual")) return MonitorConditionOperator.LessThanOrEqual;
+            if (StringComparer.OrdinalIgnoreCase.Equals(value, "Equal")) return MonitorConditionOperator.EqualsValue;
```
Comment (ArcturusZhang) [inline sdk/monitor/Azure.ResourceManager.Monitor/src/Customized/Models/MonitorConditionOperator.Serialization.cs:30]: also here

**Review by ArcturusZhang — APPROVED** [review]

File: sdk/monitor/Azure.ResourceManager.Monitor/src/Customized/Models/MonitorConditionOperator.Serialization.cs (hunk 1 above)
Comment (HarveyLink) [reply sdk/monitor/Azure.ResourceManager.Monitor/src/Customized/Models/MonitorConditionOperator.Serialization.cs:21]: Fixed

File: sdk/monitor/Azure.ResourceManager.Monitor/src/Customized/Models/MonitorConditionOperator.Serialization.cs (hunk 2 above)
Comment (HarveyLink) [reply sdk/monitor/Azure.ResourceManager.Monitor/src/Customized/Models/MonitorConditionOperator.Serialization.cs:32]: Fixed

**Pull request merged**